packages/realm-react/src/RealmProvider.tsx

```diff
@@ -19,8 +19,15 @@
 import React, { useEffect, useState, useRef } from "react";
 import Realm from "realm";
 import { isEqual } from "lodash";
+import { useUser } from "./UserProvider";
 
-type ProviderProps = Realm.Configuration & {
+type PartialRealmConfiguration = {
+  [P in keyof Omit<Realm.Configuration, "sync">]?: Realm.Configuration[P];
+} & {
```

Comment:
Is this different from this simpler expression?

```suggestion
type PartialRealmConfiguration = Omit<Realm.Configuration, "sync"> & {
```

Reply:
Yes, since the `?` is removed.  However this might be better:
```
type PartialRealmConfiguration = Omit<Partial<Realm.Configuration>, "sync"> & {
  sync?: Partial<Realm.SyncConfiguration>;
};
```